File: algorithms/V/Valid Anagram/Valid Anagram.js
var isAnagram = function(s, t) {
    if(s.length !== t.length) return false
    
    let charFreq = {}
    
    for(let char of s){
        charFreq[char] ? charFreq[char] +=1 : charFreq[char] = 1
    }
    
    for(let char of t){
        if(!(charFreq[char])) return false
        charFreq[char] -= 1
    }
    return true
};

File: scripts/algorithms/V/Valid Anagram/Valid Anagram.rs
// Runtime: 1 ms (Top 86.07%) | Memory: 2.30 MB (Top 35.27%)

impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
       let mut map = std::collections::HashMap::new();
       s.chars().for_each(|c| *map.entry(c).or_insert(0) += 1); 
       t.chars().for_each(|c| *map.entry(c).or_insert(0) -= 1); 
       !map.into_values().any(|count| count != 0)
    }
}
